arch/src/aarch64/layout.rs

```diff
@@ -88,6 +95,13 @@ pub const PCI_MMIO_CONFIG_SIZE_PER_SEGMENT: u64 = 4096 * 256;
 /// Start of RAM.
 pub const RAM_START: GuestAddress = GuestAddress(0x4000_0000);
 
+/// 32-bit reserved area: 64MiB before 4GiB
+pub const MEM_32BIT_RESERVED_START: GuestAddress = GuestAddress(0xfc00_0000);
+pub const MEM_32BIT_RESERVED_SIZE: u64 = 0x0400_0000;
+
+/// Start of 64-bit RAM.
+pub const RAM_64BIT_START: GuestAddress = GuestAddress(0x1_0000_0000);
```

Comment:
> The name `RAM_64BIT_START` is a little bit stange to me personally, simply because 4GB is not 64bit, I think x86 uses the name highmem? Same applies for the "ram 32bit", but I am totally ok with this name if you prefer this one.

I don't understand why 4GB (0x1_0000_0000) is not a 64bit address. 

This name looks clear enough to stand for the address beyond 32-bit space. And it has been long used, see: https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/arch/src/x86_64/layout.rs#L120

Reply:
> I don't understand why 4GB (0x1_0000_0000) is not a 64bit address.

Simply because 0x100000000=100000000000000000000000000000000 (33bits), but that's fine.

Reply:
` 64bit address` is the counterpart concept to `32bit address`. It doesn't mean the top bit has to be 1.

Reply:
> ` 64bit address` is the counterpart concept to `32bit address`. It doesn't mean the top bit has to be 1.

Yeah I agree with that, what I am trying to improve here is the readability of the code. We know the concept of "`64bit address` is the counterpart concept to `32bit address`.", but to be honest this name does not really reflect this concept (at least not really to me...I am not sure if this is my problem), and therefore in my opinion, if someone read the cloud hypervisor arm code for the first time, he might get confused. But yeah if you like this name just keep it.

Reply:
Sorry, I still didn't get the point why "this name does not really reflect this concept". Can you suggest the naming? We can discuss something specific.

Reply:
highmem and lowmem concept just like x86? Not really sure if this is suitable for arm though.

Reply:
HIGHMEM of x86 points to 1 MiB. AFAIK, there is some historical reason for that name. And arm didn't share the same reason. I don't think it's a good name.

"64bit" has nothing with the physical address size (40/48/52).

Putting all the discussion so far, and we are sharing the same naming for the same thing on both architectures, I still think this is a proper name.